sdk/servicebus/service-bus/test/serviceBusClient.spec.ts

```diff
@@ -98,7 +99,11 @@ describe("Errors with non existing Namespace #RunInBrowser", function(): void {
   });
 
   const testError = (err: Error): void => {
-    should.equal(err.name, "ServiceCommunicationError", "ErrorName is different than expected");
+    should.equal(
+      (err as MessagingError).code,
```

Comment:
Nit: Consider using the `MessagingError` as the type in the call back signature to avoid the casting.

`const testError = (err: MessagingError): void => {`

Reply:
If you consider making this change, then please re-trigger the live/integration test run

Reply:
This had been explored earlier - would conflict with the error handler function type expected of `registerMessageHandler`, hence wasn't possible to do that. 

Looks like we should likely update the user provided error handler signature to accept both Error and MessagingError? based on latest core-amqp changes cc: @chradek

Reply:
@ramya0820 Yes exactly! We likewise updated the error handler for event hubs to accept both Error and MessagingError, and will need to do the same in Service Bus.

We could have a helper function in the tests if we really want to avoid all the casting, maybe something like:
```typescript
function isMessagingError(err: any): err is MessagingError {
  return err.name === "MessagingError";
}
```
And then we can throw an error in our tests too if the error is or isn't a MessagingError based on what we expect.

Reply:
> Looks like we should likely update the user provided error handler signature to accept both Error and MessagingError? based on latest core-amqp changes

Yes.

Reply:
> Looks like we should likely update the user provided error handler signature to accept both Error and MessagingError? based on latest core-amqp changes

Yes.

Reply:
Like the isMessagingError check - added it in.
All tests pass - https://dev.azure.com/azure-sdk/internal/_build/results?buildId=233856
Triggering another round post latest changes